api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionsUriHelper.java

```diff
@@ -10,13 +12,17 @@
 public class SubscriptionsUriHelper {
 
     public static PaginationLinks.Link createSubscriptionListLink(
-            final Optional<String> owningApplication, final Set<String> eventTypes, final int offset,
+            final Optional<String> owningApplication, final Set<String> eventTypes,
+            final Optional<AuthorizationAttribute> reader, final int offset,
             final Optional<SubscriptionTokenLister.Token> token, final int limit, final boolean showStatus) {
 
         final UriComponentsBuilder urlBuilder = UriComponentsBuilder.fromPath("/subscriptions");
         if (!eventTypes.isEmpty()) {
             urlBuilder.queryParam("event_type", eventTypes.toArray());
         }
+        reader.ifPresent(readerOpt -> {
```

Comment:
Not sure I really follow the naming here: `readerOpt` is no longer optional, as we know it's present, but OK.

Reply:
Actually, it is optional because on SubscriptionRepository I am not doing a method overload as in EventTypeRepository, hence passing a nullable all the way down.

The reason for that is the implementation complexity would increase.